js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx

```diff
@@ -93,6 +94,20 @@ export const AppProvider = (props: AppProviderProps) => {
     [headerObject, websocketURI],
   );
 
+  const retryLink = React.useMemo(() => {
+    return new RetryLink({
+      attempts: {
+        max: 2,
```

Comment:
Little scared about how this can  multiply OOMing requests but the max should keep it from being that crazy. Just a second attempt will likely resolve a lot of flakes so i think the reward is greater than the risk. Maybe even max 3. Curious on @gibsondan 's take

Reply:
Yeah I was seeing a 504 error code for the oom earlier.
<img width="1412" alt="Screenshot 2024-06-27 at 1 25 47 PM" src="https://github.com/dagster-io/dagster/assets/2286579/709e01d6-9fbf-483c-bd1f-4918d54ddf94">

Reply:
I was imagining a single retry to start to capture the case where a single pod OOMs would be the right balance here - not strongly held

Reply:
Hmm late to the party but I wonder if we should retry only queries and not mutations? I wonder if there are cases where a 500X mutation could succeed and then error, and retrying would perform an action again?